src/main/java/org/jenkinsci/plugins/workflow/support/actions/LogActionImpl.java

```diff
@@ -149,12 +153,10 @@ public FlowNode getParent() {
      */
     private static boolean isRunning(FlowNode node) {
         if (node instanceof BlockStartNode) {
-            for (FlowNode head : node.getExecution().getCurrentHeads()) {
-                if (new LinearBlockHoppingScanner().findFirstMatch(head, Predicates.equalTo(node)) != null) {
-                    return true;
-                }
-            }
-            return false;
+            // Block start is considered running if currently executing nodes are part of the block
+            List<FlowNode> headNodes = node.getExecution().getCurrentHeads();
+            AbstractFlowScanner scanner = (headNodes.size() > 1) ? new DepthFirstScanner() : new LinearScanner();
```

Comment:
If there is just one head, `LinearBlockHoppingScanner` would be more efficient than `LinearScanner`, right? So you do not need to examine unrelated previous blocks.

`DepthFirstScanner` is not going to be terribly efficient for a big graph, but I suppose it will at least be faster than the current code.